src/backend/distributed/transaction/multi_shard_transaction.c

```diff
@@ -28,61 +30,42 @@ static void RegisterShardPlacementXactCallback(void);
 
 
 /* Global variables used in commit handler */
-static List *shardPlacementConnectionList = NIL;
-static bool isXactCallbackRegistered = false;
+static HTAB *shardConnectionHash = NULL;
+static shmem_startup_hook_type prev_shmem_startup_hook = NULL;
 
 
 /*
- * OpenTransactionsToAllShardPlacements opens connections to all placements of
- * the given shard Id Pointer List and returns the hash table containing the connections.
- * The resulting hash table maps shardIds to ShardConnection structs.
+ * OpenTransactionsToAllShardPlacements opens connections to all placements
+ * using the provided shard identifier list. Connections accumulate in a global
+ * shardConnectionHash variable for use (and re-use) within this transaction.
  */
-HTAB *
+void
 OpenTransactionsToAllShardPlacements(List *shardIntervalList, char *userName)
```

Comment:
Since we may not necessarily open connections, should we just call it something like: GetTransactionsForAllShardPlacements?